src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -0,0 +1,155 @@
+package scala
+package collection
+package immutable
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+
+import scala.collection.mutable.{Builder, ImmutableBuilder}
+import scala.annotation.unchecked.{uncheckedVariance => uV}
+import scala.collection.immutable.Map.useBaseline
+
+/** This class implements immutable maps using a vector/map-based data structure, which preserves insertion order.
+  *
+  *  Unlike `ListMap`, `VectorMap` has amortized effectively constant lookup at the expense
+  *  of using extra memory and generally lower performance for other operations
+  *
+  *  @tparam K      the type of the keys contained in this vector map.
+  *  @tparam V      the type of the values associated with the keys in this vector map.
+  *
+  * @author Matthew de Detrich
+  * @version 2.13
+  * @since 2.13
+  * @define coll immutable vector map
+  * @define Coll `immutable.VectorMap`
+  */
+final class VectorMap[K, +V] private[immutable] (
+    val fields: Vector[K],
+    val underlying: Map[K, (Int, V)])
```

Comment:
please make the fields private - they are not part of the public interface

Reply:
Woops missed this, thanks for picking it up, will fix it tonight